reader_concurrency_semaphore.cc

```diff
@@ -1124,6 +1128,11 @@ reader_concurrency_semaphore::~reader_concurrency_semaphore() {
 reader_concurrency_semaphore::inactive_read_handle reader_concurrency_semaphore::register_inactive_read(mutation_reader reader,
         const dht::partition_range* range) noexcept {
     auto& permit = reader.permit();
+    if (permit->aborted()) {
```

Comment:
You need to cover the same case in `unregister_inactive_read`. The permit's timer is not canceled here, so it's possible it times out while the read is registered as inactive.

Reply:
Readers that time-out while registered as inactive are evicted, we already handle this case.